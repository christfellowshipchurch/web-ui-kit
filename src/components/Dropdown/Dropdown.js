import React, { useState } from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {
  faCheckCircle,
  faAngleUp,
  faAngleDown,
  faCalendarAlt,
  faBan
} from "@fortawesome/pro-light-svg-icons";


const Dropdown = ({ title, items }) => {

  const [isHover, setisHover] = useState(false);
  const [isOpen, setIsOpen] = useState(false);

  let blueHover = isHover || isOpen ? "#00AEEF" : "#8d8d8e";

  let arrow = isOpen ? faAngleUp : faAngleDown;

  const menuClass = `dropdown-menu py-0 dropdown-menu border-0 ${
    isOpen ? " show" : ""
  }`;

  return (
    <div className="dropdown">
      {/* Button */}
      <div
        className="d-flex text-left border-bottom align-items-center"
        id="dropdownMenuButton"
        data-toggle="dropdown"
        aria-haspopup="true"
        aria-expanded="false"
        onMouseEnter={() => setisHover(!isHover)}
        onMouseLeave={() => setisHover(!isHover)}
        onClick={() => setIsOpen(!isOpen)}
      >
        <FontAwesomeIcon
          icon={faCalendarAlt}
          size="2x"
          className="p-1"
          color={blueHover}
        />
        <button className="btn text-muted">{title}</button>

        <FontAwesomeIcon
          icon={arrow}
          color="#757575"
          size="3x"
          className="p-1 ml-auto mr-1"
        />
      </div>
      {/* Menu */}
      <div className={menuClass} aria-labelledby="dropdownMenuButton">
        {/* Item */}

        {items.map((n, i) => {
          return (
            n.disable
            ? (
              <div
              key={i}
              className="d-flex align-items-center dropdown-item border-bottom pr-0"
              href={n.action}
            >
              <p 
                className="mb-0 pl-3 disable"
              >
                {n.call}
              </p>
              <FontAwesomeIcon
                icon={faBan}
                color={'lightgrey'}
                className="ml-auto p-1"
                size="2x"
              />
            </div>
            )
            : (<div
              key={i}
              className="d-flex align-items-center dropdown-item border-bottom text-muted pr-0"
              href={n.action}
            >
              <p className="mb-0 pl-3">{n.call}</p>
              <FontAwesomeIcon
                icon={faCheckCircle}
                className="ml-auto p-1 checkbox"
                size="2x"
              />
            </div>)
          );
        })}
      </div>
    </div>
  );
};
export default Dropdown;
