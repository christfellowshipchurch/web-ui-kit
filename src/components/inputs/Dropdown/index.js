import React, { useState } from 'react'
import PropTypes from 'prop-types'
import { get } from 'lodash'
import {
  faExclamationCircle,
  faCheckCircle,
  faBan,
  faUser,
  faAngleDown,
  faAngleUp
} from "@fortawesome/pro-light-svg-icons"

import InputContainer from '../inputContainer'
import InputIcon from '../inputIcon'

const Dropdown = ({
  label,
  helper,
  placeholder,
  disabled,
  description,
  error,
  success,
  onChange,
  withSuccess,
  value,
  icon,
  options,
  ...selectProps
}) => {
  const [focused, setFocused] = useState(false)
  const [hovering, setHovering] = useState(false)
  const [inputValue, setInputValue] = useState(value)
  const color = (focused || hovering) && !disabled
    ? '#00aeef'
    : disabled ? '#e6e6e6' : '#525252'
  const suffix = focused ? faAngleUp : faAngleDown


  return (
    <InputContainer
      label={label}
      placeholder={placeholder}
      description={error && !disabled ? error : description}
      focused={focused || !!inputValue}
      hasValue={true}
      prefix={<InputIcon icon={icon} color={color} />}
      suffix={suffix && <InputIcon icon={suffix} color={color} />}
    >
      <select
        value={value}
        className="w-100 py-0"
        onFocus={() => setFocused(true)}
        onBlur={() => setFocused(false)}
        onMouseEnter={() => setHovering(true)}
        onMouseLeave={() => setHovering(false)}
        onChange={(e) => { setInputValue(e.target.value); onChange(e) }}
        disabled={disabled}
        {...selectProps}
      >
        {options.map((n, i) => {
          const value = get(n, 'value', n)
          const label = get(n, 'label', n)

          return (
            <option
              value={value}
              key={i}
            >
              {label}
            </option>
          )
        })}
      </select>
    </InputContainer>
  )
}

Dropdown.defaultProps = {
  onChange: () => { },
  icon: faUser,
  options: []
}

Dropdown.propTypes = {
  onChange: PropTypes.func,
  icon: PropTypes.object,

  // Options allow an array of:
  //  strings: where the value and label are the same
  //  objects: { value, label } to distinguish the label from the value
  options: PropTypes.arrayOf(
    PropTypes.oneOfType(
      PropTypes.shape({
        value: PropTypes.string,
        label: PropTypes.string
      }),
      PropTypes.string
    )),
}

export default Dropdown