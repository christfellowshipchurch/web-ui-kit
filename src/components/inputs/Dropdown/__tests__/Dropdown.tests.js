import React from 'react'
import renderer from 'react-test-renderer'
import {
    configure, mount,
} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Dropdown from '../index'

configure({ adapter: new Adapter() })

describe('Dropdown', () => {
    it('renders a Dropdown without crashing', () => {
        const tree = renderer.create(
            <Dropdown
                label="Label"
                description="Description" />
        )

        expect(tree).toMatchSnapshot()
    })

    it('renders a Dropdown with a placeholder', () => {
        const tree = renderer.create(
            <Dropdown
                label="Label"
                description="Description"
                placeholder="Placeholder" />
        )

        expect(tree).toMatchSnapshot()
    })

    it('renders a Dropdown with a prefix', () => {
        const tree = renderer.create(
            <Dropdown
                label="Label"
                description="Description"
                prefix={<div style={{ background: 'red', height: 20, width: 20 }}></div>} />
        )

        expect(tree).toMatchSnapshot()
    })

    it('renders a Dropdown with a suffix', () => {
        const tree = renderer.create(
            <Dropdown
                label="Label"
                description="Description"
                suffix={<div style={{ background: 'red', height: 20, width: 20 }}></div>} />
        )

        expect(tree).toMatchSnapshot()
    })

    it('renders a Dropdown with an error', () => {
        const tree = renderer.create(
            <Dropdown
                label="Label"
                description="Description"
                error="Here's an error" />
        )

        expect(tree).toMatchSnapshot()
    })

    it('renders a Dropdown with using success', () => {
        const tree = renderer.create(
            <Dropdown
                label="Label"
                description="Description"
                withSuccess />
        )

        expect(tree).toMatchSnapshot()
    })

    it('renders a disabled Dropdown', () => {
        const tree = renderer.create(
            <Dropdown
                label="Label"
                description="Description"
                disabled />
        )

        expect(tree).toMatchSnapshot()
    })

    it('simulates the click of a Dropdown', () => {
        const wrapper = mount(
            <Dropdown
                label="Label"
                description="Description" />
        )
        const select = wrapper.find('select')

        // Finds the select and clicks it
        select.simulate('focus')
        wrapper.update()

        expect(wrapper).toMatchSnapshot()

        // Removes focus
        select.simulate('blur')
        wrapper.update()

        expect(wrapper).toMatchSnapshot()
    })

    it('simulates the mouse enter and leave of a Dropdown', () => {
        const wrapper = mount(
            <Dropdown
                label="Label"
                description="Description" />
        )
        const select = wrapper.find('select')

        // Mouse Enters
        select.simulate('mouseenter')
        wrapper.update()

        expect(wrapper).toMatchSnapshot()

        // Mouse leaves
        select.simulate('mouseleave')
        wrapper.update()

        expect(wrapper).toMatchSnapshot()
    })

    it('simulates the mouse enter and leave of a Dropdown', () => {
        const wrapper = mount(
            <Dropdown
                label="Label"
                description="Description" />
        )
        const select = wrapper.find('select')

        // Types in the select
        select.simulate('change', { target: { value: 'My new value' } })
        wrapper.update()

        expect(wrapper).toMatchSnapshot()
    })
})