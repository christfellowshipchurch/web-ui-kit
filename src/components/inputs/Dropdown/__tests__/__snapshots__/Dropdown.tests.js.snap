// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown renders a Dropdown with a placeholder 1`] = `
<div>
  <div
    className="d-flex flex-row align-items-end border-bottom border-light pb-2 px-1"
  >
    <div
      className="d-flex align-items-end justify-content-center"
      style={
        Object {
          "fontSize": 24,
          "minWidth": 30,
        }
      }
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-user fa-w-14 fa-1x "
        color="#525252"
        data-icon="user"
        data-prefix="fal"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M313.6 288c-28.7 0-42.5 16-89.6 16-47.1 0-60.8-16-89.6-16C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4zM416 464c0 8.8-7.2 16-16 16H48c-8.8 0-16-7.2-16-16v-41.6C32 365.9 77.9 320 134.4 320c19.6 0 39.1 16 89.6 16 50.4 0 70-16 89.6-16 56.5 0 102.4 45.9 102.4 102.4V464zM224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm0-224c52.9 0 96 43.1 96 96s-43.1 96-96 96-96-43.1-96-96 43.1-96 96-96z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className="w-100 form-group mb-0 pt-3 px-2 p-relative"
    >
      <label
        className="m-0 input-label-sm"
      >
        Label
      </label>
      <select
        className="w-100 py-0"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
    </div>
    <div
      className="d-flex align-items-end justify-content-center"
      style={
        Object {
          "fontSize": 24,
          "minWidth": 30,
        }
      }
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-angle-down fa-w-8 fa-1x "
        color="#525252"
        data-icon="angle-down"
        data-prefix="fal"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 256 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M119.5 326.9L3.5 209.1c-4.7-4.7-4.7-12.3 0-17l7.1-7.1c4.7-4.7 12.3-4.7 17 0L128 287.3l100.4-102.2c4.7-4.7 12.3-4.7 17 0l7.1 7.1c4.7 4.7 4.7 12.3 0 17L136.5 327c-4.7 4.6-12.3 4.6-17-.1z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
  </div>
  <small
    className="form-text text-muted"
  >
    Description
  </small>
</div>
`;

exports[`Dropdown renders a Dropdown with a prefix 1`] = `
<div>
  <div
    className="d-flex flex-row align-items-end border-bottom border-light pb-2 px-1"
  >
    <div
      className="d-flex align-items-end justify-content-center"
      style={
        Object {
          "fontSize": 24,
          "minWidth": 30,
        }
      }
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-user fa-w-14 fa-1x "
        color="#525252"
        data-icon="user"
        data-prefix="fal"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M313.6 288c-28.7 0-42.5 16-89.6 16-47.1 0-60.8-16-89.6-16C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4zM416 464c0 8.8-7.2 16-16 16H48c-8.8 0-16-7.2-16-16v-41.6C32 365.9 77.9 320 134.4 320c19.6 0 39.1 16 89.6 16 50.4 0 70-16 89.6-16 56.5 0 102.4 45.9 102.4 102.4V464zM224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm0-224c52.9 0 96 43.1 96 96s-43.1 96-96 96-96-43.1-96-96 43.1-96 96-96z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className="w-100 form-group mb-0 pt-3 px-2 p-relative"
    >
      <label
        className="m-0 input-label-sm"
      >
        Label
      </label>
      <select
        className="w-100 py-0"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        prefix={
          <div
            style={
              Object {
                "background": "red",
                "height": 20,
                "width": 20,
              }
            }
          />
        }
      />
    </div>
    <div
      className="d-flex align-items-end justify-content-center"
      style={
        Object {
          "fontSize": 24,
          "minWidth": 30,
        }
      }
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-angle-down fa-w-8 fa-1x "
        color="#525252"
        data-icon="angle-down"
        data-prefix="fal"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 256 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M119.5 326.9L3.5 209.1c-4.7-4.7-4.7-12.3 0-17l7.1-7.1c4.7-4.7 12.3-4.7 17 0L128 287.3l100.4-102.2c4.7-4.7 12.3-4.7 17 0l7.1 7.1c4.7 4.7 4.7 12.3 0 17L136.5 327c-4.7 4.6-12.3 4.6-17-.1z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
  </div>
  <small
    className="form-text text-muted"
  >
    Description
  </small>
</div>
`;

exports[`Dropdown renders a Dropdown with a suffix 1`] = `
<div>
  <div
    className="d-flex flex-row align-items-end border-bottom border-light pb-2 px-1"
  >
    <div
      className="d-flex align-items-end justify-content-center"
      style={
        Object {
          "fontSize": 24,
          "minWidth": 30,
        }
      }
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-user fa-w-14 fa-1x "
        color="#525252"
        data-icon="user"
        data-prefix="fal"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M313.6 288c-28.7 0-42.5 16-89.6 16-47.1 0-60.8-16-89.6-16C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4zM416 464c0 8.8-7.2 16-16 16H48c-8.8 0-16-7.2-16-16v-41.6C32 365.9 77.9 320 134.4 320c19.6 0 39.1 16 89.6 16 50.4 0 70-16 89.6-16 56.5 0 102.4 45.9 102.4 102.4V464zM224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm0-224c52.9 0 96 43.1 96 96s-43.1 96-96 96-96-43.1-96-96 43.1-96 96-96z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className="w-100 form-group mb-0 pt-3 px-2 p-relative"
    >
      <label
        className="m-0 input-label-sm"
      >
        Label
      </label>
      <select
        className="w-100 py-0"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        suffix={
          <div
            style={
              Object {
                "background": "red",
                "height": 20,
                "width": 20,
              }
            }
          />
        }
      />
    </div>
    <div
      className="d-flex align-items-end justify-content-center"
      style={
        Object {
          "fontSize": 24,
          "minWidth": 30,
        }
      }
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-angle-down fa-w-8 fa-1x "
        color="#525252"
        data-icon="angle-down"
        data-prefix="fal"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 256 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M119.5 326.9L3.5 209.1c-4.7-4.7-4.7-12.3 0-17l7.1-7.1c4.7-4.7 12.3-4.7 17 0L128 287.3l100.4-102.2c4.7-4.7 12.3-4.7 17 0l7.1 7.1c4.7 4.7 4.7 12.3 0 17L136.5 327c-4.7 4.6-12.3 4.6-17-.1z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
  </div>
  <small
    className="form-text text-muted"
  >
    Description
  </small>
</div>
`;

exports[`Dropdown renders a Dropdown with an error 1`] = `
<div>
  <div
    className="d-flex flex-row align-items-end border-bottom border-light pb-2 px-1"
  >
    <div
      className="d-flex align-items-end justify-content-center"
      style={
        Object {
          "fontSize": 24,
          "minWidth": 30,
        }
      }
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-user fa-w-14 fa-1x "
        color="#525252"
        data-icon="user"
        data-prefix="fal"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M313.6 288c-28.7 0-42.5 16-89.6 16-47.1 0-60.8-16-89.6-16C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4zM416 464c0 8.8-7.2 16-16 16H48c-8.8 0-16-7.2-16-16v-41.6C32 365.9 77.9 320 134.4 320c19.6 0 39.1 16 89.6 16 50.4 0 70-16 89.6-16 56.5 0 102.4 45.9 102.4 102.4V464zM224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm0-224c52.9 0 96 43.1 96 96s-43.1 96-96 96-96-43.1-96-96 43.1-96 96-96z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className="w-100 form-group mb-0 pt-3 px-2 p-relative"
    >
      <label
        className="m-0 input-label-sm"
      >
        Label
      </label>
      <select
        className="w-100 py-0"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
    </div>
    <div
      className="d-flex align-items-end justify-content-center"
      style={
        Object {
          "fontSize": 24,
          "minWidth": 30,
        }
      }
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-angle-down fa-w-8 fa-1x "
        color="#525252"
        data-icon="angle-down"
        data-prefix="fal"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 256 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M119.5 326.9L3.5 209.1c-4.7-4.7-4.7-12.3 0-17l7.1-7.1c4.7-4.7 12.3-4.7 17 0L128 287.3l100.4-102.2c4.7-4.7 12.3-4.7 17 0l7.1 7.1c4.7 4.7 4.7 12.3 0 17L136.5 327c-4.7 4.6-12.3 4.6-17-.1z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
  </div>
  <small
    className="form-text text-muted"
  >
    Here's an error
  </small>
</div>
`;

exports[`Dropdown renders a Dropdown with using success 1`] = `
<div>
  <div
    className="d-flex flex-row align-items-end border-bottom border-light pb-2 px-1"
  >
    <div
      className="d-flex align-items-end justify-content-center"
      style={
        Object {
          "fontSize": 24,
          "minWidth": 30,
        }
      }
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-user fa-w-14 fa-1x "
        color="#525252"
        data-icon="user"
        data-prefix="fal"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M313.6 288c-28.7 0-42.5 16-89.6 16-47.1 0-60.8-16-89.6-16C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4zM416 464c0 8.8-7.2 16-16 16H48c-8.8 0-16-7.2-16-16v-41.6C32 365.9 77.9 320 134.4 320c19.6 0 39.1 16 89.6 16 50.4 0 70-16 89.6-16 56.5 0 102.4 45.9 102.4 102.4V464zM224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm0-224c52.9 0 96 43.1 96 96s-43.1 96-96 96-96-43.1-96-96 43.1-96 96-96z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className="w-100 form-group mb-0 pt-3 px-2 p-relative"
    >
      <label
        className="m-0 input-label-sm"
      >
        Label
      </label>
      <select
        className="w-100 py-0"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
    </div>
    <div
      className="d-flex align-items-end justify-content-center"
      style={
        Object {
          "fontSize": 24,
          "minWidth": 30,
        }
      }
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-angle-down fa-w-8 fa-1x "
        color="#525252"
        data-icon="angle-down"
        data-prefix="fal"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 256 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M119.5 326.9L3.5 209.1c-4.7-4.7-4.7-12.3 0-17l7.1-7.1c4.7-4.7 12.3-4.7 17 0L128 287.3l100.4-102.2c4.7-4.7 12.3-4.7 17 0l7.1 7.1c4.7 4.7 4.7 12.3 0 17L136.5 327c-4.7 4.6-12.3 4.6-17-.1z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
  </div>
  <small
    className="form-text text-muted"
  >
    Description
  </small>
</div>
`;

exports[`Dropdown renders a Dropdown without crashing 1`] = `
<div>
  <div
    className="d-flex flex-row align-items-end border-bottom border-light pb-2 px-1"
  >
    <div
      className="d-flex align-items-end justify-content-center"
      style={
        Object {
          "fontSize": 24,
          "minWidth": 30,
        }
      }
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-user fa-w-14 fa-1x "
        color="#525252"
        data-icon="user"
        data-prefix="fal"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M313.6 288c-28.7 0-42.5 16-89.6 16-47.1 0-60.8-16-89.6-16C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4zM416 464c0 8.8-7.2 16-16 16H48c-8.8 0-16-7.2-16-16v-41.6C32 365.9 77.9 320 134.4 320c19.6 0 39.1 16 89.6 16 50.4 0 70-16 89.6-16 56.5 0 102.4 45.9 102.4 102.4V464zM224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm0-224c52.9 0 96 43.1 96 96s-43.1 96-96 96-96-43.1-96-96 43.1-96 96-96z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className="w-100 form-group mb-0 pt-3 px-2 p-relative"
    >
      <label
        className="m-0 input-label-sm"
      >
        Label
      </label>
      <select
        className="w-100 py-0"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
    </div>
    <div
      className="d-flex align-items-end justify-content-center"
      style={
        Object {
          "fontSize": 24,
          "minWidth": 30,
        }
      }
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-angle-down fa-w-8 fa-1x "
        color="#525252"
        data-icon="angle-down"
        data-prefix="fal"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 256 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M119.5 326.9L3.5 209.1c-4.7-4.7-4.7-12.3 0-17l7.1-7.1c4.7-4.7 12.3-4.7 17 0L128 287.3l100.4-102.2c4.7-4.7 12.3-4.7 17 0l7.1 7.1c4.7 4.7 4.7 12.3 0 17L136.5 327c-4.7 4.6-12.3 4.6-17-.1z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
  </div>
  <small
    className="form-text text-muted"
  >
    Description
  </small>
</div>
`;

exports[`Dropdown renders a disabled Dropdown 1`] = `
<div>
  <div
    className="d-flex flex-row align-items-end border-bottom border-light pb-2 px-1"
  >
    <div
      className="d-flex align-items-end justify-content-center"
      style={
        Object {
          "fontSize": 24,
          "minWidth": 30,
        }
      }
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-user fa-w-14 fa-1x "
        color="#e6e6e6"
        data-icon="user"
        data-prefix="fal"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M313.6 288c-28.7 0-42.5 16-89.6 16-47.1 0-60.8-16-89.6-16C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4zM416 464c0 8.8-7.2 16-16 16H48c-8.8 0-16-7.2-16-16v-41.6C32 365.9 77.9 320 134.4 320c19.6 0 39.1 16 89.6 16 50.4 0 70-16 89.6-16 56.5 0 102.4 45.9 102.4 102.4V464zM224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm0-224c52.9 0 96 43.1 96 96s-43.1 96-96 96-96-43.1-96-96 43.1-96 96-96z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className="w-100 form-group mb-0 pt-3 px-2 p-relative"
    >
      <label
        className="m-0 input-label-sm"
      >
        Label
      </label>
      <select
        className="w-100 py-0"
        disabled={true}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
    </div>
    <div
      className="d-flex align-items-end justify-content-center"
      style={
        Object {
          "fontSize": 24,
          "minWidth": 30,
        }
      }
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-angle-down fa-w-8 fa-1x "
        color="#e6e6e6"
        data-icon="angle-down"
        data-prefix="fal"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 256 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M119.5 326.9L3.5 209.1c-4.7-4.7-4.7-12.3 0-17l7.1-7.1c4.7-4.7 12.3-4.7 17 0L128 287.3l100.4-102.2c4.7-4.7 12.3-4.7 17 0l7.1 7.1c4.7 4.7 4.7 12.3 0 17L136.5 327c-4.7 4.6-12.3 4.6-17-.1z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
  </div>
  <small
    className="form-text text-muted"
  >
    Description
  </small>
</div>
`;

exports[`Dropdown simulates the click of a Dropdown 1`] = `ReactWrapper {}`;

exports[`Dropdown simulates the click of a Dropdown 2`] = `ReactWrapper {}`;

exports[`Dropdown simulates the mouse enter and leave of a Dropdown 1`] = `ReactWrapper {}`;

exports[`Dropdown simulates the mouse enter and leave of a Dropdown 2`] = `ReactWrapper {}`;

exports[`Dropdown simulates the mouse enter and leave of a Dropdown 3`] = `ReactWrapper {}`;
